### PR TITLE
dev/core#1541 Fix ICalendar random invalid utf8 (PHP <7.4 compat) and tests

### PR DESCRIPTION
**Pull request by mlutfy**
Overview
----------------------------------------

`mb_str_split` is new in PHP 7.4. This PR uses the mb function only if available (using the meh-convention found elsewhere in core). Also add tests, sort of. Follow-up to #23840.

Technical Details
----------------------------------------

`unformatText`  does not seem to be used other than in tests, but I found that it was not passing tests for long lines, since when lines are split (newline and space), then they were not unformatted correctly.

Also, the test admittedly does not check for broken unicode, because I guess it could be split/merged correctly.

Comments
----------------------------------------

sorry @agileware-justin for breaking your pending PR again.

### CODE REVIEW

**Comment by agileware-justin** [comment]
Thanks @mlutfy - all good

Impacts https://github.com/civicrm/civicrm-core/pull/23808

File: CRM/Utils/ICalendar.php [hunk 1]
```diff
@@ -51,6 +53,7 @@ public static function formatText($text) {
    * @return string
    */
   public static function unformatText($text) {
+    $text = str_replace("\n ", "", $text);
```
Comment (demeritcowboy) [inline CRM/Utils/ICalendar.php:56]: Agree that universe suggests this is only used in tests.

In fact, only ICalendarTest, so moving this function into that file might make sense. But I'll wait until Justin rebases the other PR first since that will be funnier. (grin)
